t_sparse_list.py

```diff
@@ -109,6 +109,17 @@ def test_slice_reversal_empty(self):
         sl = sparse_list.SparseList(4)
         self.assertEquals([None, None, None, None], sl[::-1])
 
+    def test_slice_with_list_read(self):
+        sl = sparse_list.SparseList([1, 2, 3, 4, 5])
+        ip = [0, 2, 4]
+        self.assertEquals([1, 3, 5], sl[ip])
```

Comment:
The use-case is from `numpy.array` : [integer array indexing](http://docs.scipy.org/doc/numpy/reference/arrays.indexing.html#integer-array-indexing). Scientists use this a lot to vectorize their operations and gain performance by leveraging numpy's internals.